processing/src/main/java/io/druid/segment/StringDimensionMergerV9.java

```diff
@@ -167,18 +168,18 @@ public void writeMergedValueMetadata(List<IndexableAdapter> adapters) throws IOE
 
     cardinality = 0;
     if (numMergeIndex > 1) {
-      IndexMerger.DictionaryMergeIterator iterator = new IndexMerger.DictionaryMergeIterator(dimValueLookups, true);
+      dictionaryMergeIterator = new IndexMerger.DictionaryMergeIterator(dimValueLookups, true);
```

Comment:
The iterator itself doesn't, but the buffers are used later (see my other comment)